### PR TITLE
Add documentation for the coc.nvim integration (cf. #1349)

### PR DESCRIPTION
**Pull request by voldikss**

### CODE REVIEW

File: doc/vimtex.txt [hunk 1]
```diff
@@ -2892,12 +2893,31 @@ Note: If you want to change the default value of `TEXMFHOME` in your shell
 AUTOCOMPLETE                                             *vimtex-complete-auto*
 
 Vim does not provide automatic completion by itself, but there exist at least
-four good plugins that provide this: ||deoplete|, |neocomplete|, |ncm2|,
+several good plugins that provide this: |coc.nvim|, ||deoplete|, |neocomplete|, |ncm2|,
```
Comment (lervag) [inline doc/vimtex.txt]: While you're at it, could you fix the erroneous `||deoplete|` -> `|deoplete|`?

File: doc/vimtex.txt [hunk 2]
```diff
@@ -2892,12 +2893,31 @@ Note: If you want to change the default value of `TEXMFHOME` in your shell
...
+coc.nvim~
+                                                     *vimtex-complete-coc.nvim*
+|coc.nvim| is an intellisense engine for Vim8 & Neovim. It's a completion 
+framework and language server client which supports extension features of VSCode.
+The project is here: https://github.com/neoclide/coc.nvim.
+
+|coc-vimtex| is the plugin which provides `vimtex` integration for |coc.nvim|.
+see https://github.com/neoclide/coc-vimtex. To use it, just run >
+
+  :CocInstall coc-vimtex
```
Comment (lervag) [inline doc/vimtex.txt]: If I use vim-plug: Is it not better to install it in the "usual" way?

**Comment by lervag** [comment]
Thanks! A couple of very minor comments. I'll merge when these are resolved.

File: doc/vimtex.txt (hunk 2 above)
Comment (voldikss) [reply doc/vimtex.txt]: Actually coc.nvim is not that similar to other plugins such as deoplete and ncm2.
coc.nvim is a framework which is mainly developed using `typescript`, so many coc's plugin can not be installed using usual package manager(vim-plug, vundle...). It provides its own command `CocInstall` and `CocUninstall` to install and uninstall the plugins

Comment (voldikss) [reply doc/vimtex.txt]: Oh, I forgot it! 

coc.nvim is installed using vim-plug (I'll added it soon)

after coc.nvim is installed, coc-vimtex must be install using `CocInstall`

Comment (lervag) [reply doc/vimtex.txt]: Hmm. How do I setup my vimrc so that a given extension is automatically installed, then, if missing?

Comment (voldikss) [reply doc/vimtex.txt]: Sorry I didn't understand what you mean by "automatically installed" and "missing" (I'm not good at English...)
Do you mean something like `Plug 'neoclide/coc.nvim`, { 'do' : 'call CocInstall command to installed coc-vimtex' }

File: doc/vimtex.txt (hunk 1 above)
Comment (voldikss) [reply doc/vimtex.txt]: Okay

File: doc/vimtex.txt (hunk 2 above)
Comment (lervag) [reply doc/vimtex.txt]: Ok, no problem. I think it is OK like this. If I should use `coc.nvim` I'll refine things myself if I find it necessary.

Comment (voldikss) [reply doc/vimtex.txt]: Thank you!

**Pull request merged**